io-amqp1_0-impl/src/main/java/org/apache/pulsar/ecosystem/io/amqp/Connection.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.ecosystem.io.amqp;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+/**
+ * Connection with a list of URI's and whether the failover prefix should be used.
+ */
+@Data
+@Accessors(chain = true)
+public class Connection {
+
+    private boolean useFailover;
+    private List<ConnectionUri> uris;
+
+    public String getUri(){
```

Comment:
nit: `joining` accepts prefix and suffix parameters ([doc](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#joining-java.lang.CharSequence-java.lang.CharSequence-java.lang.CharSequence-)). Maybe you can use `if-else` make the code more clear.

This is not a hard request but personal bias.

Reply:
thanks, have changed it.